code/game/objects/items/robot/robot_parts.dm

```diff
@@ -112,132 +207,73 @@
-
-			O.job = "Cyborg"
-
-			O.cell = chest.cell
-			O.cell.loc = O
-			W.loc = O//Should fix cybros run time erroring when blown up. It got deleted before, along with the frame.
-
-			// Since we "magically" installed a cell, we also have to update the correct component.
-			if(O.cell)
-				var/datum/robot_component/cell_component = O.components["power cell"]
-				cell_component.wrapped = O.cell
-				cell_component.installed = 1
-
-			feedback_inc("cyborg_birth",1)
-			O.Namepick()
-
-			del(src)
-		else
-			user << "\blue The MMI must go in after everything else!"
-
+	// HANDLE RENAME
 	if (istype(W, /obj/item/weapon/pen))
 		var/t = stripped_input(user, "Enter new robot name", src.name, src.created_name, MAX_NAME_LEN)
 		if (!t)
 			return
 		if (!in_range(src, usr) && src.loc != usr)
 			return
-
 		src.created_name = t
+	// HANDLE PART ASSEMBLY
+	if(istype(W,/obj/item/robot_parts))
+		var/obj/item/robot_parts/part = W
+		part.attach_to_robot(user,src)
+		return
+	// HANDLE ROBOT CREATION
+	if(istype(W, /obj/item/device/mmi))
+		var/obj/item/device/mmi/brain = W
+		if (allowed_to_build(user,brain)) // we are allowed to build this robot
+			user.drop_item() // drop this thing
+			if (src.head.law_computer) // do we have a law computer? If so, we're making a standard robot
+				create_robot(brain)
+			else // otherwise we're making a shell
+				create_shell(brain)
+				
+				
+/obj/item/robot_parts/robot_suit/proc/create_robot(obj/item/device/mmi/brain as obj)
+	var/mob/living/silicon/robot/new_robot = new(get_turf(loc), unfinished = 1)
+	if(!new_robot) // something has gone poorly
+		return
+	// move the brain
+	new_robot.mmi = brain 
+	new_robot.invisibility = 0
+	new_robot.custom_name = created_name
+	new_robot.updatename("Default")
+	brain.brainmob.mind.transfer_to(new_robot) // shove that mob in the mmi
+	if(new_robot.mind && new_robot.mind.special_role)
+		new_robot.mind.store_memory("In case you look at this after being borged, the objectives are only here until I find a way to make them not show up for you, as I can't simply delete them without screwing up round-end reporting. --NeoFite")
+	new_robot.job = "Cyborg"
+	new_robot.cell = chest.cell
+	new_robot.cell.loc = new_robot
+	brain.loc = new_robot //Should fix cybros run time erroring when blown up. It got deleted before, along with the frame.
+	if(new_robot.cell) // Since we "magically" installed a cell, we also have to update the correct component.
+		var/datum/robot_component/cell_component = new_robot.components["power cell"]
+		cell_component.wrapped = new_robot.cell
+		cell_component.installed = 1
+	feedback_inc("cyborg_birth",1)
+	new_robot.Namepick()
+	del(src)
+	
+	
+/obj/item/robot_parts/robot_suit/proc/create_shell(obj/item/device/mmi/brain as obj)
+	var/mob/living/carbon/human/machine/new_shell = new(src.loc)
+	var/key=brain.brainmob.mind.key
```

Comment:
What is this? It reports as being unused. Can delete/comment out, or does it actually serve a purpose?